build/storage/tests/it/ptf_tests/system_tools/test_platform.py

```diff
@@ -96,62 +155,171 @@ def _install_docker(self):
+        super().__init__(SSHTerminal(StorageTargetConfig()), cmd_sender)
+        self.docker.add_container("storage_target_container", StorageTargetContainer())
+
+    def create_subsystem(self, nqn: str, port_to_expose: int, storage_target_port: int):
+        return self.cmd_sender.create_subsystem(
+            self.get_ip_address(),
+            nqn,
+            port_to_expose,
+            storage_target_port,
+        )
+
+    def create_ramdrives(self, ramdrives_number, port, nqn, spdk_port):
+        nvme_storages = []
+        for i in range(ramdrives_number):
+            nvme_storage = RemoteNvmeStorage(
+                self.get_ip_address(),
+                port,
+                nqn,
+                self.cmd_sender.create_ramdrive(
+                    i, self.get_ip_address(), nqn, spdk_port
+                ),
+            )
+            nvme_storages.append(nvme_storage)
+        return nvme_storages
 
 
 class IPUStoragePlatform(BaseTestPlatform):
-    def __init__(self):
-        config = IPUStorageConfig()
-        terminal = SSHTerminal(config)
-        super().__init__(terminal)
+    def __init__(self, cmd_sender=None):
+        super().__init__(SSHTerminal(IPUStorageConfig()), cmd_sender)
+        self.docker.add_container("ipu_storage_container", IPUStorageContainer())
+
+    @property
+    def sma_port(self):
+        return self.config.sma_port
+
+    def create_virtio_blk_devices(
+        self,
+        host_target_address_service,
+        volumes,
+        physical_ids,
+    ):
+        device_handles = []
+        for volume, physical_id in zip(volumes, physical_ids):
+            device_handles.append(
+                VirtioBlkDevice(
+                    self.cmd_sender.create_virtio_blk_device(
+                        self.get_ip_address(),
+                        host_target_address_service,
+                        volume.guid,
+                        physical_id,
+                        volume.nvme_controller_address.ip_address,
+                        volume.nvme_controller_address.port,
+                        volume.nqn,
+                        self.sma_port,
+                    ),
+                    volume,
+                    self,
+                    host_target_address_service,
+                )
+            )
+        return device_handles
+
+    def create_virtio_blk_devices_sequentially(
+        self,
+        host_target_address_service,
+        volumes,
+    ):
+        return self.create_virtio_blk_devices(
+            host_target_address_service,
+            volumes,
+            range(len(volumes)),
+        )
 
-    # TODO add implementation
-    def create_virtio_blk_device(self):
-        return "VirtioBlkDevice"
+    def delete_virtio_blk_devices(self, devices_handles):
+        for device_handle in devices_handles:
+            device_handle.delete(self.cmd_sender)
+
+    def clean(self):
+        # TODO delete all alocated devices
```

Comment:
is this not happening?

Reply:
now all containers are kill and but it nvme it is not enough